src/components/elements/input/NumberInput.tsx

```diff
@@ -61,49 +67,50 @@ const NumberInput = ({
     }
   };
 
-  // Prevent form submission on Enter. Enter should toggle the state.
-  const onKeyDown: KeyboardEventHandler<HTMLInputElement> = useCallback((e) => {
-    if (e.key.match(/(ArrowDown|ArrowUp)/)) {
-      e.preventDefault();
-    }
-  }, []);
-
-  const preventValueChangeOnScroll: WheelEventHandler<HTMLDivElement> =
-    useCallback((e) => {
-      // Prevent the input value change
-      (e.target as HTMLInputElement).blur();
+  const decimalScale = currency ? 2 : 0;
+  const prefix = currency ? '$' : undefined;
 
-      // Prevent the page/container scrolling
-      e.stopPropagation();
+  const handleChange: OnValueChange = (v) => {
+    const syntheticEvent = {
+      target: {
+        value: v.value,
+        name: props.name, // If you have a name prop
+      },
+      // Add other event properties you might need
+      preventDefault: () => {},
+      stopPropagation: () => {},
+    } as React.ChangeEvent<HTMLInputElement>;
 
-      // Refocus immediately, on the next tick (after the current
-      // function is done)
-      setTimeout(() => {
-        (e.target as HTMLInputElement).focus();
-      }, 0);
-    }, []);
+    onChange(syntheticEvent);
+  };
 
   return (
-    <TextInput
-      type='text'
+    <NumericFormat
+      error={!!(error || errorMessage)}
+      helperText={error ? undefined : errorMessage || helperText}
+      customInput={TextInput}
+      onValueChange={handleChange}
+      onBlur={handleBlur}
+      value={(value || '') as string}
+      isAllowed={withValueLimit}
+      thousandSeparator
+      decimalScale={decimalScale}
       inputProps={{
+        pattern: '[0-9]*', // hint mobile keyboards
         inputMode: 'numeric',
-        pattern: '[0-9]*',
         min,
         max,
-        onKeyDown: disableArrowKeys ? onKeyDown : undefined,
         'aria-labelledby': ariaLabelledBy,
         ...inputProps,
       }}
-      onWheel={preventValueChangeOnScroll}
-      InputProps={{ ...currencyInputProps, ...InputProps }}
-      onBlur={handleBlur}
-      value={value}
       placeholder={currency ? '0' : undefined}
-      {...props}
-      error={error || !!errorMessage}
-      // If there is a server error, show that instead of the local message
-      helperText={error ? undefined : errorMessage || props.helperText}
+      InputProps={{
+        startAdornment: prefix ? (
+          <InputAdornment position='start'>{prefix}</InputAdornment>
+        ) : undefined,
+        ...InputProps,
+      }}
+      {...(props as any)}
```

Comment:
I pushed it since I had it locally anyway: https://github.com/greenriver/hmis-frontend/pull/988/commits/f0721bb8e4b03f0c364f4b9a25b71af79f2e96a8 lmk what you think!

Reply:
that's a nice improvement, thanks!